lib/HLSL/HLExpandStoreIntrinsics.cpp

```diff
@@ -0,0 +1,136 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// HLExpandStoreIntrinsics.cpp                                               //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
```

Comment:
A little write up in the header explaining what this pass does would be very helpful.

Reply:
The comment is on the class. Since the file only defines that class, I feel like a file-level comment would be redundant and the current comment location is on what's actually being described since the class implements the pass, not the file. What do you think?

Reply:
this is totally an optional comment. also, I did notice your comment at the class level. although, conventionally, I have often seen some blurb explaining a pass in the file header, e.g.: [this](https://llvm.org/doxygen/SROA_8cpp_source.html).

Reply:
True, but then they declare 5-6 classes in that file. If you stick to one-class-per-file the file header comment becomes redundant. As it is, I don't see what I could say that wouldn't duplicate or be better added to the class comment